saleor/graphql/discount/mutations.py

```diff
@@ -270,7 +270,12 @@ class SaleUpdateMinimalVariantPriceMixin:
     def success_response(cls, instance):
         # Update the "minimal_variant_prices" of the associated, discounted
         # products (including collections and categories).
-        update_products_minimal_variant_prices_of_discount_task.delay(instance.pk)
+        if (
+            instance.products.all()
```

Comment:
Why not `instance.products.exsists()`?  I think it should be much faster. Also, why not to move this entire check to `update_products_minimal_variant_prices_of_discount_task`? Then we would be sure that it is checked in every case that you actually use this task.

Reply:
Depends whether `all()` evaluates to `exists()`, that was my initial question. I don't know if @kswiatek92 checked or not what behavior Django adopts in this case

Reply:
I didn't move this to task, because in task I only have pk, which later is used to get instance, then it checks if have any products, categories, collections, if not raise error

Reply:
@NyanKiyoshi sorry I thought it was statement not question, I checked it and `.all()` doesn't evaluates to `.exists()`